engine/src/org/pentaho/di/www/TransformationMap.java

```diff
@@ -41,18 +46,15 @@
  *
  */
 public class TransformationMap {
-  private Map<CarteObjectEntry, Trans> transformationMap;
-  private Map<CarteObjectEntry, TransConfiguration> configurationMap;
+  private final Map<CarteObjectEntry, TransData> transMap;
 
-  private Map<String, List<SocketPortAllocation>> hostServerSocketPortsMap;
+  private final Map<String, List<SocketPortAllocation>> hostServerSocketPortsMap;
```

Comment:
Instead of a <String, List>, I would rather you used a <String, Deque<SocketPortAllocation>> with the implementation of a ConcurrentlinkedDeque. The iteration over the ConcurrentLinkedDeque will be identical, your Deque will be in the same order as added - but you will be able to do all your other operations (iteration, etc) without synchronization below.